arangod/IResearch/IResearchView.cpp

```diff
@@ -846,6 +847,8 @@ IResearchView::IResearchView(
         ReadMutex mutex(_mutex); // 'store' can be asynchronously modified
         SCOPED_LOCK(mutex);
 
+        const auto readerBeforeSync = store->_reader;
```

Comment:
will that copy the reader, or be a reference? if it copies, is that copy expensive?

Reply:
just wondering because I cannot figure out the type that quickly.

Reply:
reader is just a pimpl, so the copy is cheap and noexcept